src/frontends/pytorch/src/op/native_multi_head_attention.cpp

```diff
@@ -0,0 +1,192 @@
+    const auto value_proj_bias =
+        context.mark_node(std::make_shared<opset10::Slice>(qkv_bias, ev_2_slice_1d, ev_3_slice_1d, one_1d, zero_1d));
+
+    const auto query_weighted =
+        context.mark_node(std::make_shared<opset10::MatMul>(query, query_proj_weight, false, true));
+    const auto key_weighted = context.mark_node(std::make_shared<opset10::MatMul>(key, key_proj_weight, false, true));
+    const auto value_weighted =
+        context.mark_node(std::make_shared<opset10::MatMul>(value, value_proj_weight, false, true));
+
+    const auto query_biased = context.mark_node(std::make_shared<opset10::Add>(query_weighted, query_proj_bias));
+    const auto key_biased = context.mark_node(std::make_shared<opset10::Add>(key_weighted, key_proj_bias));
+    const auto value_biased = context.mark_node(std::make_shared<opset10::Add>(value_weighted, value_proj_bias));
+
+    const auto qkv_reshape_dims = context.mark_node(
+        std::make_shared<opset10::Concat>(OutputVector{batch_size, seq_size, heads_1d, neg_one_1d}, 0));
+    const auto qv_transpose_dims = context.mark_node(opset10::Constant::create(element::i64, Shape{4}, {0, 2, 1, 3}));
+    const auto k_transpose_dims = context.mark_node(opset10::Constant::create(element::i64, Shape{4}, {0, 2, 3, 1}));
+
+    const auto query_reshaped =
+        context.mark_node(std::make_shared<opset10::Reshape>(query_biased, qkv_reshape_dims, false));
+    const auto key_reshaped =
+        context.mark_node(std::make_shared<opset10::Reshape>(key_biased, qkv_reshape_dims, false));
+    const auto value_reshaped =
+        context.mark_node(std::make_shared<opset10::Reshape>(value_biased, qkv_reshape_dims, false));
+
+    const auto query_transposed =
+        context.mark_node(std::make_shared<opset10::Transpose>(query_reshaped, qv_transpose_dims));
+    const auto key_transposed = context.mark_node(std::make_shared<opset10::Transpose>(key_reshaped, k_transpose_dims));
+    const auto value_transposed =
+        context.mark_node(std::make_shared<opset10::Transpose>(value_reshaped, qv_transpose_dims));
+
+    const auto scale_one = context.mark_node(std::make_shared<opset10::ConvertLike>(one_1d, query_transposed));
+    const auto scale_dim = context.mark_node(std::make_shared<opset10::ConvertLike>(embed_div_heads, query_transposed));
+    const auto scale_dim_sqrt = context.mark_node(std::make_shared<opset10::Sqrt>(scale_dim));
+    const auto scale = context.mark_node(std::make_shared<opset10::Divide>(scale_one, scale_dim_sqrt));
+
+    const auto query_key_transpose_dot_product =
+        context.mark_node(std::make_shared<opset10::MatMul>(query_transposed, key_transposed));
+
+    auto scaled_dot_product =
+        context.mark_node(std::make_shared<opset10::Multiply>(query_key_transpose_dot_product, scale));
+
+    // Mask handling
+    if (!context.input_is_none(12)) {
+        auto atten_mask = context.get_input(9);
```

Comment:
What if `mask_type` is provided, but `mask` is None? Looks like you also need to verify that.

Reply:
Added - I thought it was an impossible case in a pre-built graph